### PR TITLE
refactor codes

### PR DESCRIPTION
**Pull request by minggo**
Re: https://github.com/cocos/cocos-engine/issues/17715

- use self to reduce package size
- simplify codes

### CODE REVIEW

**Comment by minggo** [comment]
@cocos-robot run test cases

File: cocos/ui/scroll-view.ts [hunk 1]
```diff
@@ -1310,10 +1347,9 @@ export class ScrollView extends ViewGroup {
     }
 
     protected _flattenVectorByDirection (vector: Vec3): Vec3 {
-        const result = vector;
-        result.x = this.horizontal ? result.x : 0;
-        result.y = this.vertical ? result.y : 0;
-        return result;
+        if (!this.horizontal) vector.x = 0;
+        if (!this.vertical) vector.y = 0;
+        return vector;
```
Comment (minggo) [inline cocos/ui/scroll-view.ts:1353]: Simplify codes.

File: cocos/ui/scroll-view.ts [hunk 2]
```diff
@@ -1536,96 +1580,100 @@ export class ScrollView extends ViewGroup {
             _tempVec3_1.set(_tempVec2_1.x, _tempVec2_1.y, 0);
             uiTransformComp.convertToNodeSpaceAR(_tempVec3, _tempVec3);
             uiTransformComp.convertToNodeSpaceAR(_tempVec3_1, _tempVec3_1);
-            Vec3.subtract(vec, _tempVec3, _tempVec3_1);
+            Vec3.subtract(out, _tempVec3, _tempVec3_1);
```
Comment (minggo) [inline cocos/ui/scroll-view.ts:1584]: Just use out instead, don't have to create a new temple object vec.

File: cocos/ui/scroll-view.ts [hunk 3]
```diff
@@ -1188,16 +1226,14 @@ export class ScrollView extends ViewGroup {
         if (captureListeners) {
             // captureListeners are arranged from child to parent
             for (const listener of captureListeners) {
-                const item = listener;
-
-                if (this.node === item) {
+                if (this.node === listener) {
```
Comment (minggo) [inline cocos/ui/scroll-view.ts:1229]: item is not needed, just delete it.

File: cocos/ui/scroll-view.ts [hunk 4]
```diff
@@ -1240,13 +1276,12 @@ export class ScrollView extends ViewGroup {
...
         let factor = targetDelta.length() / originalMoveLength;
-        targetDelta.add(deltaMove);
 
         if (this.brake > 0 && factor > 7) {
             factor = Math.sqrt(factor);
-            const clonedDeltaMove = deltaMove.clone();
-            clonedDeltaMove.multiplyScalar(factor);
-            targetDelta.set(clonedDeltaMove);
+            targetDelta.set(deltaMove);
+            targetDelta.multiplyScalar(factor + 1);
```
Comment (minggo) [inline cocos/ui/scroll-view.ts:1283]:
```typescript
// clonedDeltaMove = deltaMove
const clonedDeltaMove = deltaMove.clone(); 
// clonedDeltaMove = clonedDeltaMove * factor = deltaMove * factor
clonedDeltaMove.multiplyScalar(factor); 
// targetDelta = clonedDeltaMove = deltaMove * factor          
targetDelta.set(clonedDeltaMove);     
// targetDelta = targetDelta + deltaMove = deltaMove * factor + deltaMove = deltaMove * (factor + 1)              
targetDelta.add(deltaMove);                              
```

File: cocos/ui/scroll-view.ts [hunk 5]
```diff
@@ -1240,13 +1276,12 @@ export class ScrollView extends ViewGroup {
 
         const originalMoveLength = deltaMove.length();
         let factor = targetDelta.length() / originalMoveLength;
-        targetDelta.add(deltaMove);
```
Comment (minggo) [inline cocos/ui/scroll-view.ts:1243]: as targetDelta will be set in `if (this.brake > 0 && factor > 7)`, so move it to `else` block bellow.

File: cocos/ui/scroll-view.ts (hunk 4 above)
Comment (qiuguohua) [reply cocos/ui/scroll-view.ts:1283]: Why does the factor need to be increased?

Comment (minggo) [reply cocos/ui/scroll-view.ts:1283]: Please refer to the comment above.

**Review by qiuguohua — APPROVED** [review]

**Pull request merged**